src/Search/SimilarDocumentsSearchResult.php

```diff
@@ -0,0 +1,133 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Meilisearch\Search;
+
+class SimilarDocumentsSearchResult implements \Countable, \IteratorAggregate
+{
+    /**
+     * @var array<int, array<string, mixed>>
+     */
+    private array $hits;
+
+    /**
+     * `estimatedTotalHits` is the attributes returned by the Meilisearch server
+     * and its value will not be modified by the methods in this class.
+     * Please, use `hitsCount` if you want to know the real size of the `hits` array at any time.
+     */
+    private int $estimatedTotalHits;
+    private int $hitsCount;
+    private int $offset;
+    private int $limit;
+    private int $processingTimeMs;
+
+    private string $id;
+
+    public function __construct(array $body)
+    {
+        $this->id = $body['id'];
+        $this->hits = $body['hits'];
+        $this->hitsCount = \count($body['hits']);
+        $this->processingTimeMs = $body['processingTimeMs'];
+        $this->offset = $body['offset'];
+        $this->limit = $body['limit'];
+        $this->estimatedTotalHits = $body['estimatedTotalHits'];
+    }
+
+    /**
+     * Return a new {@see SearchResult} instance.
```

Comment:
I don't see how this returns a new SearchResult, just modifies previous one :) it should clone itself then..
Also what's the point of this method, if you can directly call transformHits?

Reply:
i will remove this method as the `transformHits` is public as well. I will add documentation on `transformHits` specifying what it is doing.

Also, is the `transformHits` needed or should the user write his own custom function to manipulate the search results ?

Reply:
I'd say not needed, unless we have a use case for this in the library itself.. butnas I understand you just copied from another class so maybe @curquiza should decide

Reply:
If you think it's not needed, let's keep it simple, and remove it